app/Libraries/Search/ScoreSearch.php

```diff
@@ -95,4 +70,50 @@ public function getQuery(): BoolQuery
 
         return $query;
     }
+
+    private function addModsFilter(BoolQuery $query): void
+    {
+        $mods = $this->params->mods;
+        if ($mods === null || count($mods) === 0) {
+            return;
+        }
+
+        $modsHelper = app('mods');
+        $allMods = new Set(array_keys($modsHelper->mods[$this->params->rulesetId]));
```

Comment:
just noticed 'rulesetId' should default to something or there should be some `null` check.

Reply:
To be fixed by #9129